src/struct_info.rs

```diff
@@ -609,7 +605,10 @@ impl<'a> StructInfo<'a> {
             ),
             proc_macro2::Span::call_site(),
         );
-        let early_build_error_message = format!("Missing required field {}", field_name);
+        let mut early_build_error_message = format!("Missing required field {}", field_name);
+        if field.builder_attr.setter.extend.is_some() {
+            write!(&mut early_build_error_message, " (single item setter: {})", field.item_name()).unwrap();
+        }
```

Comment:
There probably should be a special message in case neither of these are available as initialisers.
I'll stop for today, but I've added this point to my to-do list.

(As an aside, I've just been able to confirm this feature will work for my use-case 🥳
I've queued that feature up just so it's easier to keep track of, but that won't change how I work on this one.)